From 77c2a30769d251b00d357fd17c9ed26230215e68 Mon Sep 17 00:00:00 2001
From: Yimin Peng <quic_yiminp@quicinc.com>
Date: Tue, 8 Aug 2023 16:32:28 +0800
Subject: [PATCH 1/3] Add compatible string instance for VFIO platform

Enable a comptabile string for the VFIO platform driver. Fix
unexpected exit in probe.

Change-Id: I8f72f5112efcdac104db212aed79b1cb3c495167
Signed-off-by: Venkatakrishnaiah Pari <quic_vpari@quicinc.com>
Signed-off-by: Yimin Peng <quic_yiminp@quicinc.com>
---
 drivers/vfio/platform/vfio_platform.c        | 8 +++++++-
 drivers/vfio/platform/vfio_platform_common.c | 7 +++++--
 drivers/vfio/vfio_main.c                     | 3 +--
 3 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/drivers/vfio/platform/vfio_platform.c b/drivers/vfio/platform/vfio_platform.c
index 82cedcebfd90..6eeff94a90d3 100644
--- a/drivers/vfio/platform/vfio_platform.c
+++ b/drivers/vfio/platform/vfio_platform.c
@@ -16,7 +16,7 @@
 #define DRIVER_AUTHOR   "Antonios Motakis <a.motakis@virtualopensystems.com>"
 #define DRIVER_DESC     "VFIO for platform devices - User Level meta-driver"
 
-static bool reset_required = true;
+static bool reset_required = false;
 module_param(reset_required, bool, 0444);
 MODULE_PARM_DESC(reset_required, "override reset requirement (default: 1)");
 
@@ -108,11 +108,17 @@ static const struct vfio_device_ops vfio_platform_ops = {
 	.mmap		= vfio_platform_mmap,
 };
 
+static const struct of_device_id qcom_vfio_platform_of_match[] = {
+	{ .compatible = "qcom,vfio-platform"},
+	{ }
+};
+
 static struct platform_driver vfio_platform_driver = {
 	.probe		= vfio_platform_probe,
 	.remove		= vfio_platform_remove,
 	.driver	= {
 		.name	= "vfio-platform",
+		.of_match_table = qcom_vfio_platform_of_match,
 	},
 	.driver_managed_dma = true,
 };
diff --git a/drivers/vfio/platform/vfio_platform_common.c b/drivers/vfio/platform/vfio_platform_common.c
index 1a0a238ffa35..7325ff463cf0 100644
--- a/drivers/vfio/platform/vfio_platform_common.c
+++ b/drivers/vfio/platform/vfio_platform_common.c
@@ -650,10 +650,13 @@ int vfio_platform_init_common(struct vfio_platform_device *vdev)
 	mutex_init(&vdev->igate);
 
 	ret = vfio_platform_get_reset(vdev);
-	if (ret && vdev->reset_required)
+	if (ret && vdev->reset_required) {
 		dev_err(dev, "No reset function found for device %s\n",
 			vdev->name);
-	return ret;
+		return ret;
+	}
+
+	return 0;
 }
 EXPORT_SYMBOL_GPL(vfio_platform_init_common);
 
diff --git a/drivers/vfio/vfio_main.c b/drivers/vfio/vfio_main.c
index 6e8804fe0095..bce879c5ebba 100644
--- a/drivers/vfio/vfio_main.c
+++ b/drivers/vfio/vfio_main.c
@@ -511,8 +511,7 @@ static struct vfio_group *vfio_group_find_or_alloc(struct device *dev)
 	 * valid for cases where we are using iommu groups.
 	 */
 	if (!device_iommu_capable(dev, IOMMU_CAP_CACHE_COHERENCY)) {
-		iommu_group_put(iommu_group);
-		return ERR_PTR(-EINVAL);
+		dev_warn(dev, "No IOMMU_CAP_CACHE_COHERENCY support?\n");
 	}
 
 	group = vfio_group_get_from_iommu(iommu_group);
-- 
2.17.1

